swift/Samples/TabBarContainer/Sources/TabBarScreenViewController.swift

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright 2019 Square Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import Workflow
+import WorkflowUI
+
+internal final class TabBarScreenContainerViewController<Content: Screen>: ScreenViewController<TabBarScreen<Content>>, UITabBarDelegate {
+    private var containerView: ContainerView
+    private var contentViewController: DescribedViewController
+    
+    private let tabBar: UITabBar
+    
+    required init(screen: TabBarScreen<Content>, environment: ViewEnvironment) {
+        self.containerView = ContainerView()
+        
+        self.contentViewController = DescribedViewController(
+            screen: screen.currentScreen,
+            environment: environment
+        )
+        
+        self.tabBar = UITabBar()
+        
+        super.init(screen: screen, environment: environment)
+        
+        self.tabBar.delegate = self
+    }
+
+    override internal func screenDidChange(from previousScreen: TabBarScreen<Content>, previousEnvironment: ViewEnvironment) {
+        update()
+    }
+
+    private func update() {
+        contentViewController.update(screen: screen.currentScreen, environment: environment)
+        
+        var selectedTabBarItem: UITabBarItem?
+        
+        let tabBarItems = screen.barItems.enumerated().map { (index, barItem) -> UITabBarItem in
+            let tabBarItem = UITabBarItem(
+                title: barItem.title,
+                image: barItem.image,
+                selectedImage: barItem.selectedImage
+            )
+            
+            tabBarItem.badgeValue = barItem.badge
+            
+            if index == screen.selectedIndex {
+                selectedTabBarItem = tabBarItem
+            }
+            
+            return tabBarItem
+        }
+        
+        tabBar.setItems(tabBarItems, animated: false)
+        tabBar.selectedItem = selectedTabBarItem
+    }
+
+    override internal func viewDidLoad() {
+        super.viewDidLoad()
+
+        view.addSubview(containerView)
+        view.addSubview(tabBar)
+
+        addChild(contentViewController)
+        containerView.contentView.addSubview(contentViewController.view)
```

Comment:
Why containerView instead of adding the contentViewController’s view directly?

Reply:
I received feedback on my SplitScreen PR from @AquaGeek to use a container view to codify a view who sets it's subviews bounds to its frame on every layout pass. Decided to keep the pattern the same.

Reply:
Can’t find that discussion from that PR. But I feel like this adds an unnecessary layer to the code.

Reply:
https://github.com/square/workflow/pull/925#discussion_r376552966

Reply:
For context, IIRC that discussion was around adding static container subviews in the split screen so that you don't have to redo all the sizing math when swapping out contents — you just hand the new content view to the appropriate container and it adds it as a subview and does a `view.frame = bounds`.